src/charm.py

```diff
@@ -190,55 +170,53 @@ def ensure_state(self):
         )
 
     def handle_ingress(self):
-        ingress = self._get_interface("ingress")
-
-        if ingress:
-            for app_name, version in ingress.versions.items():
-                data = {
-                    "prefix": "/dex",
-                    "rewrite": "/dex",
-                    "service": self.model.app.name,
-                    "port": self.model.config["port"],
-                }
+        interface = self._get_interface("ingress")
+
+        if not interface:
+            return
+
+        data = {
+            "prefix": "/dex",
+            "rewrite": "/dex",
+            "service": self.model.app.name,
+            "port": self.model.config["port"],
+        }
 
-                ingress.send_data(data, app_name)
+        interface.send_data(data)
 
     def _check_leader(self):
         if not self.unit.is_leader():
             # We can't do anything useful when not the leader, so do nothing.
-            raise CheckFailedError("Waiting for leadership", WaitingStatus)
+            raise ErrorWithStatus("Waiting for leadership", WaitingStatus)
 
     def _check_container_connection(self):
         if not self._container.can_connect():
-            raise CheckFailedError("Waiting for pod startup to complete", WaitingStatus)
+            raise ErrorWithStatus("Waiting for pod startup to complete", WaitingStatus)
 
     def _get_interface(self, interface_name):
         # Remove this abstraction when SDI adds .status attribute to NoVersionsListed,
         # NoCompatibleVersionsListed:
         # https://github.com/canonical/serialized-data-interface/issues/26
         try:
-            try:
-                interface = get_interface(self, interface_name)
-            except NoVersionsListed as err:
-                raise CheckFailedError(str(err), WaitingStatus)
-        except CheckFailedError as err:
-            self.logger.debug("_get_interface ~ Checkfailederror catch")
-            self.model.unit.status = err.status
-            self.logger.info(str(err.status))
-            return
+            interface = get_interface(self, interface_name)
+        except NoVersionsListed as err:
+            raise ErrorWithStatus(str(err), WaitingStatus)
```

Comment:
should also have the `NoCompatibleVersions` catch
```suggestion
        except NoVersionsListed as err:
            raise ErrorWithStatus(str(err), WaitingStatus)
        except NoCompatibleVersions as err:
            raise ErrorWithStatus(str(err), BlockedStatus)
```